### PR TITLE
Create working dir if it does not exist

### PR DESCRIPTION
**Pull request by rylev**
Currently we error if the working directory does not exist already.

### CODE REVIEW

**Review by itowlson — CHANGES_REQUESTED** [review]
I can't remember what discussions we had at the time about whether implicit directory creation was good (so convenient) or bad (what if they meant `~/myapps` but typoed `~/maypps`) - other people may have views on this...

**Comment by vdice** [comment]
E2e tests should now run successfully, just need to rebase.  Though, it looks like there is a failure in the Go SDK tests as well.

File: src/commands/up.rs [hunk 1]
```diff
@@ -157,6 +153,22 @@ impl UpCommand {
...
+        };
+        if !working_dir_holder.path().exists() {
+            std::fs::create_dir_all(working_dir_holder.path())
+                .context("Could not create working directory")?;
+        }
+        let working_dir = working_dir_holder
+            .path()
+            .canonicalize()
+            .context("Could not canonicalize working directory")?;
+        Ok(working_dir)
```
[the remark this answers is not in the document]
Comment (rylev) [reply src/commands/up.rs:169]: Very good catch! I'll change that and make a comment.

**Comment by rylev** [comment]
@itowlson the issue around the temp directory has been fixed. How do you want to make a call on whether auto creation of the working dir is the right way to go?

**Review by itowlson — APPROVED** [review]
LGTM - thanks for adding the comment.

Re whether the behaviour is desirable or not, I can't find any record of the alleged previous discussion.  I don't have strong feelings either way, so I guess either wield your maintainer powers or do a quick poll in Discord?

**Comment by mikkelhegn** [comment]
LGTM

**Comment by rylev** [comment]
Note to future selves - this was discussed by several folks including @mikkelhegn and @melissaklein24, and everyone seemed relatively happy with this small change.

**Pull request merged**